mouse/mouse.nim

```diff
@@ -42,113 +51,82 @@ when defined(windows):
-        elif `type` == "absolute":
-            SetCursorPos(x, y)
+        of Absolute:
+            SetCursorPos(int32 x, int32 y)
 
-    proc smoothMove*(x: int, y: int, smoothingStep: float, `sleep` = 3, `type` = Absolute) = 
+    proc smoothMove*(x: int, y: int, smoothingStep: float, sleep = 3, typ = Absolute) = 
         var currPos: Point
         GetCursorPos(currPos)
 
-        if `type` == "relative":
-            var x,y = calcRelative((int(currPos.x), int(currPos.y)), (x,y))
+        var
+            x = x
+            y = y
+        if typ == Relative:
+            (x, y) = calcRelative((int(currPos.x), int(currPos.y)), (x,y))
 
         var i = 0.0
         while i < 1.0:
             var vec = lerp((int(currPos.x), int(currPos.y)), (x, y), i)
-            debugEcho vec
 
             SetCursorPos(int32(vec.x), int32(vec.y))
 
-            sleep(`sleep`)
+            sleep(sleep)
             i += smoothingStep
 
     proc scroll*(amount: int, orientation = Vertical, globalDelta = 120) =
         var finalAmount: int = amount * globalDelta
-        mouse_event(DWORD(orientation), 0, 0, DWORD(finalAmount), 0)
+        mouse_event(DWORD(ScrollOrientationToEvent[orientation]), 0, 0, DWORD(finalAmount), 0)
 
     proc isPressed*(button: int32): bool =
-        return bool(GetAsyncKeyState(button))
+        bool(GetAsyncKeyState(button))
 
     proc getPos*: Point =
-        var pos: Point
-        GetCursorPos(pos)
-
-        return pos
+        discard GetCursorPos(result)
 
 when defined(linux):
-    import strutils
-
-
-    proc validateButton(button: string or int): string = # bad code, probably will rewrite later
-        if button.type == string:
-            try:
-                case parseInt(button):
-                    of 1 or 2 or 3:
-                        return button
-                    else:
-                        return "invalid"
-            except:
-                case lower(button):
-                    of "left":
-                        return 1
-                    of "right":
-                        return 2
-                    of "middle":
-                        return 3
-                    else:
-                        return "invalid"
-
-        elif button.type == int:
-            case button:
-                of 1 or 2 or 3:
-                     return button
-                else:
-                    return "invalid"
-
-        else:
-            return "invalid"
-
-
-
-
-    template click*(button: string or int): string =
-        var button = validateButton(button)
-        fmt"xdo button_press -k {button};xdo button_release -k {button};"
-
-    template press*(button: string or int): string =
-        var button = validateButton(button)
-        fmt"xdo button_press -k {button}"
-
-    template release*(button: string or int): string =
-        var button = validateButton(button)
-        fmt"xdo button_release -k {button}"
+    proc buttonToXdotool(button: MouseButton): string =
```

Comment:
We just take the enum value and add 1 to it so that 0 for Left becomes 1 (Nim enums start at 0), Right becomes 2, and Middle becomes 3.